### PR TITLE
Apply jitter to watch timeout

### PR DESCRIPTION
**Pull request by trustin**
Related: #217

Motivation:

A Central Dogma server can get a large number of watch requests if
clients use the same watch timeout value, a.k.a thundering herd problem.

Modifications:

- Apply 20% jitter to all watch timeouts
- Miscellaneous
  - Reject zero or negative timeout values explicitly

Result:

Less spikes in server load

### CODE REVIEW

File: server/src/main/java/com/linecorp/centraldogma/server/internal/api/WatchService.java [hunk 1]
```diff
@@ -61,11 +64,29 @@
         return result;
     }
 
+    /**
+     * Awaits and retrieves the latest revision of the commit that changed the file that matches the specified
+     * {@link Query} since the specified {@code lastKnownRevision}. This will wait until the specified
+     * {@code timeoutMillis} passes. If there's no change during the time, the returned future will be
+     * exceptionally completed with the {@link CancellationException}.
+     */
+    public <T> CompletableFuture<Entry<T>> watchFile(Repository repo, Revision lastKnownRevision,
```
Comment (trustin) [inline server/src/main/java/com/linecorp/centraldogma/server/internal/api/WatchService.java]: This was just moved up.

File: server/src/main/java/com/linecorp/centraldogma/server/internal/api/converter/WatchRequestConverter.java [hunk 2]
```diff
@@ -63,16 +63,25 @@ public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage re
...
+            return rejectPreferHeader(preferHeader);
         }
+
+        if (timeoutSeconds <= 0) {
+            return rejectPreferHeader(preferHeader);
+        }
+        return TimeUnit.SECONDS.toMillis(timeoutSeconds);
+    }
+
+    private static long rejectPreferHeader(String preferHeader) {
```
Comment (minwoox) [inline server/src/main/java/com/linecorp/centraldogma/server/internal/api/converter/WatchRequestConverter.java]: private static void?

Comment (minwoox) [reply server/src/main/java/com/linecorp/centraldogma/server/internal/api/converter/WatchRequestConverter.java]: Ah! didn't think about that.

**Comment by trustin** [comment]
Thanks for reviewing.

**Pull request merged**